### PR TITLE
add open liberty package

### PR DESCRIPTION
**Pull request by cpanato**
xref: https://github.com/chainguard-dev/image-requests/issues/4078

### CODE REVIEW

**Comment by jamonation** [comment]
@EyeCantCU did it look like this would be buildable with OpenJ9?

**Review by jamonation — DISMISSED** [review]

**Comment by EyeCantCU** [comment]
Upstream builds with 17 and distributes images with all LTS Java releases at runtime - we'll handle this at the image level due to this

**Comment by jamonation** [comment]
What happens with adding `-x test` to the gradle commands? Not that we want to do that, but does it get things building? If it does, we could narrow down the failing tests and get them sorted out or excluded if they don't run properly in CI.

File: open-liberty.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    tag-filter: gm-
+
+test:
+  environment:
+    contents:
+      packages:
+        - openjdk-21-default-jvm
+  pipeline:
+    - runs: |
+        export PATH="/usr/share/java/openliberty/wlp/bin:$PATH"
```
Comment (jamonation) [inline open-liberty.yaml:85]:
```suggestion
        export PATH="/usr/share/java/open-liberty/wlp/bin:$PATH"
```

File: open-liberty.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    with:
+      expected-commit: daad205e2831f1d46bdb56781f75a0f17eaf9e3d
+      repository: https://github.com/OpenLiberty/open-liberty
+      tag: gm-${{package.version}}
+
+  - working-directory: dev
+    runs: |
+      ./gradlew --refresh-dependencies --stacktrace cnf:initialize -x test
+      ./gradlew --refresh-dependencies --stacktrace assemble -x test
+      ./gradlew --refresh-dependencies --stacktrace releaseNeeded -x test
```
Comment (jamonation) [inline open-liberty.yaml:40]:
```suggestion
      ./gradlew --refresh-dependencies --stacktrace cnf:initialize
      ./gradlew --refresh-dependencies --stacktrace assemble
      ./gradlew --refresh-dependencies --stacktrace releaseNeeded
```

Comment (jamonation) [reply open-liberty.yaml:40]: probably don't need javadocs here, but leaving as is

File: open-liberty.yaml [hunk 3]
```diff
@@ -0,0 +1,91 @@
+package:
```
Comment (jamonation) [inline open-liberty.yaml:1]:
```suggestion
#nolint:valid-pipeline-git-checkout-tag
package:
```

**Review by developer-guy — APPROVED** [review]

**Comment by cpanato** [comment]
🎉

**Pull request merged**